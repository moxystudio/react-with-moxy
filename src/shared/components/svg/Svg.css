@import "variables";

.svg {
    width: 24px;
    height: 24px; /* Must have an height otherwise it will have an extra space on IE11 */
    display: inline-block;
    vertical-align: middle;
    user-select: none;
    transition: fill 0.2s, fill-opacity 0.2s;
    fill: var(--color-violet);
}

.svg svg {
    width: inherit;
    height: inherit;
    display: block;  /* Remove little margin between the svg and its parent */
}
