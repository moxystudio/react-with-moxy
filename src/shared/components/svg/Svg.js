import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './Svg.css';

// See: https://github.com/Karify/external-svg-sprite-loader

const SvgInline = (props) => {
    const { svg, className, ...otherProps } = props;

    return (
        <i className={ classnames(styles.svg, className) } { ...otherProps }>
            <svg viewBox={ svg.viewBox }>
                <use xlinkHref={ svg.symbol } />
            </svg>
        </i>
    );
};

SvgInline.propTypes = {
    svg: PropTypes.object.isRequired,
    className: PropTypes.string,
};

export default SvgInline;
