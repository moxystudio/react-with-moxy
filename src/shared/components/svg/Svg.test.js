import React from 'react';
import { shallow } from 'enzyme';
import Svg from './Svg';
import cakeSvg from 'shared/media/images/icons/cake.svg';

it('should render a SVG correctly', () => {
    const tree = shallow(<Svg svg={ cakeSvg } />);

    expect(tree).toMatchSnapshot();
});

it('should render with the specified className', () => {
    const tree = shallow(<Svg svg={ cakeSvg } className="foo" />);

    expect(tree).toMatchSnapshot();
});
