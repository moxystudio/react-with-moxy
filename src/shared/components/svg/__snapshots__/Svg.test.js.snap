// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a SVG correctly 1`] = `
<i
  className="svg"
>
  <svg
    viewBox="viewBox"
  >
    <use
      xlinkHref="symbol"
    />
  </svg>
</i>
`;

exports[`should render with the specified className 1`] = `
<i
  className="svg foo"
>
  <svg
    viewBox="viewBox"
  >
    <use
      xlinkHref="symbol"
    />
  </svg>
</i>
`;
